app/src/main/java/com/github/se/travelpouch/model/notifications/NotificationContent.kt

```diff
@@ -0,0 +1,124 @@
+package com.github.se.travelpouch.model.notifications
+
+import com.github.se.travelpouch.model.Role
+
+private const val INVITER_NAME_NOT_BLANK = "Inviter name cannot be blank"
+private const val TRAVEL_TITLE_NOT_BLANK = "Travel title cannot be blank"
+
+/** Sealed class representing different types of notification content. */
+sealed class NotificationContent {
+
+  /**
+   * Data class representing an invitation notification.
+   *
+   * @property inviterName The name of the person who sent the invitation.
+   * @property travelTitle The title of the travel event.
+   * @property role The role assigned to the invitee.
+   */
+  data class InvitationNotification(
+      val inviterName: String,
+      val travelTitle: String,
+      val role: Role
+  ) : NotificationContent() {
+
+    init {
+      require(inviterName.isNotBlank()) { INVITER_NAME_NOT_BLANK }
+      require(travelTitle.isNotBlank()) { TRAVEL_TITLE_NOT_BLANK }
+    }
+
+    /**
+     * Converts the invitation notification to a display string.
+     *
+     * @return A string representation of the invitation notification.
+     */
+    override fun toDisplayString(): String {
+      return "$inviterName invited you to join the travel $travelTitle as a ${role.name}."
```

Comment:
Since role is an enum, it would be safer to check if role is not null before accessing role.name to avoid potential NullPointerException. In this case, it's not relevent to check the null because it's an enumeration that (in theory) was already checked.